op-e2e/e2eutils/disputegame/honest_helper.go

```diff
@@ -48,12 +48,14 @@ func (h *HonestHelper) StepFails(ctx context.Context, claimIdx int64, isAttack b
 	defer cancel()
 	claim := h.game.getClaim(ctx, claimIdx)
 	pos := types.NewPositionFromGIndex(claim.Position.Uint64())
-	traceIdx := pos.TraceIndex(int(h.game.MaxDepth(ctx)))
+	gameDepth := int(h.game.MaxDepth(ctx))
+	traceIdx := pos.TraceIndex(gameDepth)
```

Comment:
Need to avoid accessing the trace index here as the trace provider might need to translate it for child providers. Should be doing the same thing as the production code and just use a `StepDefend` or `MoveRight` call on Position and pass that through.